src/lib/services/code.ts

```diff
@@ -87,4 +98,81 @@ export const getCodesByAddress = async (
         offset,
       },
     })
-    .then(({ data }) => parseWithError(zCodesResponse, data));
+    .then(({ data }) => zCodesResponse.parse(data));
+
+const zCodeInfo = z
+  .object({
+    code_id: z.number().nonnegative(),
+    cw2_contract: z.string().nullable(),
+    cw2_version: z.string().nullable(),
+    hash: z.string(),
+    instantiate_permission: z.nativeEnum(AccessConfigPermission),
+    permission_addresses: z.array(zBechAddr),
+    proposal: z
+      .object({
+        id: z.number(),
+        height: z.number(),
+        created: z.string(),
+      })
+      .nullable(),
+    transaction: z
+      .object({
+        height: z.number(),
+        hash: z.string(),
+        created: z.string(),
+      })
+      .nullable(),
+    uploader: zBechAddr,
+  })
+  .transform<Partial<CodeData>>((val) => ({
+    codeId: val.code_id,
+    cw2Contract: val.cw2_contract,
+    cw2Version: val.cw2_version,
+    hash: parseTxHashOpt(val.hash),
+    instantiatePermission: val.instantiate_permission,
+    permissionAddresses: val.permission_addresses,
+    proposal: val.proposal
+      ? {
+          proposalId: val.proposal.id,
+          height: val.proposal.height,
+          created: parseDateOpt(val.proposal.created),
+        }
+      : undefined,
+    transaction: val.transaction
+      ? {
+          height: val.transaction.height,
+          hash: val.transaction.hash,
+          created: parseDateOpt(val.transaction.created),
+        }
+      : undefined,
+    uploader: val.uploader,
+  }));
+
```

Comment:
`export type Code = z.infer<typeof zCode>;`

then for every place been using type `CodeData`, change to `Code`